Existen algunas librerías personalizadas en Python un ejemplo de ellos es la
librería PyEPH, una librería desarrollada por el Instituto Nacional de Estadísticas y
Censos de la republica Argentina y cuyo fin es procesamiento en Python de las
Encuesta permanente de hogares (eph) publicadas de forma periódica. Es un
espacio donde se nuclean y centralizan los cálculos vinculados a las mismas para
posteriormente ser utilizadas en investigaciones, artículos, publicaciones, entre
otras.

Lea con cuidado las información de esta información en el siguiente link:
https://www.indec.gob.ar/ftp/cuadros/menusuperior/eph/EPH_diseno_reg_t414.pdf

In [1]:
# Siempre al principio, importamos las librerías.
import pandas as pd
import numpy as np

In [2]:
%pip install pyeph

Note: you may need to restart the kernel to use updated packages.


In [3]:
import pyeph

In [4]:
eph = pyeph.get(data="eph", year=2021, period=2, base_type='individual')

2026-06-07 18:04:28 - INFO - Descarga exitosa desde INDEC: base_individual_2021T2.zip
2026-06-07 18:04:29 - INFO - Obtenido con éxito: base_individual_2021T2.zip


### Identifique cuales son las columnas del dataset

Tiene 177 variables y 47085 filas

In [5]:
eph.columns

Index(['CODUSU', 'ANO4', 'TRIMESTRE', 'NRO_HOGAR', 'COMPONENTE', 'H15',
       'REGION', 'MAS_500', 'AGLOMERADO', 'PONDERA',
       ...
       'PDECIFR', 'ADECIFR', 'IPCF', 'DECCFR', 'IDECCFR', 'RDECCFR', 'GDECCFR',
       'PDECCFR', 'ADECCFR', 'PONDIH'],
      dtype='object', length=177)

In [6]:
eph.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47085 entries, 0 to 47084
Columns: 177 entries, CODUSU to PONDIH
dtypes: float64(103), int64(66), object(8)
memory usage: 63.6+ MB


In [7]:
eph.shape

(47085, 177)

In [8]:

nombres_columnas = eph.columns.tolist()

for columnna in nombres_columnas:
  print(columnna)

CODUSU
ANO4
TRIMESTRE
NRO_HOGAR
COMPONENTE
H15
REGION
MAS_500
AGLOMERADO
PONDERA
CH03
CH04
CH05
CH06
CH07
CH08
CH09
CH10
CH11
CH12
CH13
CH14
CH15
CH15_COD
CH16
CH16_COD
NIVEL_ED
ESTADO
CAT_OCUP
CAT_INAC
IMPUTA
PP02C1
PP02C2
PP02C3
PP02C4
PP02C5
PP02C6
PP02C7
PP02C8
PP02E
PP02H
PP02I
PP03C
PP03D
PP3E_TOT
PP3F_TOT
PP03G
PP03H
PP03I
PP03J
INTENSI
PP04A
PP04B_COD
PP04B1
PP04B2
PP04B3_MES
PP04B3_ANO
PP04B3_DIA
PP04C
PP04C99
PP04D_COD
PP04G
PP05B2_MES
PP05B2_ANO
PP05B2_DIA
PP05C_1
PP05C_2
PP05C_3
PP05E
PP05F
PP05H
PP06A
PP06C
PP06D
PP06E
PP06H
PP07A
PP07C
PP07D
PP07E
PP07F1
PP07F2
PP07F3
PP07F4
PP07F5
PP07G1
PP07G2
PP07G3
PP07G4
PP07G_59
PP07H
PP07I
PP07J
PP07K
PP08D1
PP08D4
PP08F1
PP08F2
PP08J1
PP08J2
PP08J3
PP09A
PP09A_ESP
PP09B
PP09C
PP09C_ESP
PP10A
PP10C
PP10D
PP10E
PP11A
PP11B_COD
PP11B1
PP11B2_MES
PP11B2_ANO
PP11B2_DIA
PP11C
PP11C99
PP11D_COD
PP11G_ANO
PP11G_MES
PP11G_DIA
PP11L
PP11L1
PP11M
PP11N
PP11O
PP11P
PP11Q
PP11R
PP11S
PP11T
P21
DECOCUR
IDECOCUR
RDECOCUR
GDECOCUR
PDECOCUR
ADECOC

Cree una variable llamada eph_2 que contenga las siguientes:

columnas para realizar el siguiente análisis:
 Edad (CH_06), Sexo (CH_04), Ingreso de la ocupación principal
(P21), Categoría ocupacional (CAT_OCUP), Condición de actividad
(ESTADO), Ponderación (PONDERA)

In [9]:
eph_2 = eph[['CH06','CH04','P21', 'CAT_OCUP', 'ESTADO', 'PONDERA']]

In [10]:
eph_2

,CH06,CH04,P21,CAT_OCUP,ESTADO,PONDERA
0,58,1,50000,2,1,214
1,24,1,0,3,2,214
2,51,2,0,4,1,192
3,27,1,4000,2,1,192
4,18,1,0,0,3,192
...,...,...,...,...,...,...
47080,77,2,0,0,3,86
47081,57,1,-9,2,1,86
47082,73,2,0,0,3,230
47083,44,2,18000,3,1,230


Traer un dataframe que filtre solo a las personas que tengan 21
años o menos.

In [11]:
eph_2.query('CH06<=21')

,CH06,CH04,P21,CAT_OCUP,ESTADO,PONDERA
4,18,1,0,0,3,192
8,5,1,0,0,4,250
11,19,2,0,0,3,250
12,17,1,0,0,3,250
13,14,1,0,0,3,250
...,...,...,...,...,...,...
47070,15,2,0,0,3,46
47071,13,1,0,0,3,46
47074,8,1,0,0,4,49
47077,10,1,0,0,3,111


Traer un dataframe que filtre a las personas desocupadas de
más de 40 años.

In [12]:
eph_2.query('CH06>=40 & ESTADO ==2')

,CH06,CH04,P21,CAT_OCUP,ESTADO,PONDERA
47,58,2,0,3,2,915
77,41,1,0,2,2,139
92,45,1,0,3,2,127
357,42,1,0,3,2,1086
433,48,2,0,3,2,626
...,...,...,...,...,...,...
46350,49,1,0,3,2,307
46447,72,1,0,2,2,232
46599,49,2,0,3,2,311
46715,49,2,0,3,2,215


¿Cuántas mujeres hay en la base de datos? ¿Y hombres?

#### Hombres

In [13]:
eph_2.query('CH04==1')

,CH06,CH04,P21,CAT_OCUP,ESTADO,PONDERA
0,58,1,50000,2,1,214
1,24,1,0,3,2,214
3,27,1,4000,2,1,192
4,18,1,0,0,3,192
5,45,1,120000,3,1,192
...,...,...,...,...,...,...
47077,10,1,0,0,3,111
47078,27,1,15000,2,1,82
47079,24,1,15000,2,1,82
47081,57,1,-9,2,1,86


Se utiliza shape que retorna una tupla donde estan las filas y columnas y se accede al indice que almacena la cantidad de filas que cumplen esta condicion:

In [14]:
eph_2.query('CH04==1').shape[0]

22602

### Mujeres

In [15]:
eph_2.query('CH04==2')

,CH06,CH04,P21,CAT_OCUP,ESTADO,PONDERA
2,51,2,0,4,1,192
6,28,2,12000,3,1,250
9,65,2,0,0,3,187
10,38,2,10000,3,1,250
11,19,2,0,0,3,250
...,...,...,...,...,...,...
47070,15,2,0,0,3,46
47073,41,2,57000,3,1,49
47080,77,2,0,0,3,86
47082,73,2,0,0,3,230


In [16]:
eph_2.query('CH04==2').shape[0]

24483

Por lo tanto se tienen:

**24483 mujeres** y **22602 hombres**



¿Cuál era la tasa de desocupación en el período seleccionado
para las mujeres? ¿Era más alta o más baja que la de los
hombres? (Extra: pensar el resultado usando una función)

**Nota: la tasa se calcula como Desocupados / (Ocupados + Desocupados)**

### Tasa de desocupacion para los hombres

In [17]:
eph_2.query('CH04==1 & ESTADO ==2').shape[0]/(eph_2.query('CH04==1 & ESTADO ==2').shape[0]+eph_2.query('CH04==1 & ESTADO ==1').shape[0])

0.07479247840081314

### Tasa de desocupacion para las mujeres

In [18]:
eph_2.query('CH04==2 & ESTADO ==2').shape[0]/(eph_2.query('CH04==2 & ESTADO ==2').shape[0]+eph_2.query('CH04==2 & ESTADO ==1').shape[0])

0.09144448013705964

La tasa de **desocupacion** de los **hombres era del 7.47%**, mientras que de las **mujeres era del 9.14%***. Por lo tanto, las mujeres tenian una tasa mas alta de desocupacion

## Como funcion

In [19]:
def calcular_tasa_desocupacion(df, genero):
    ocupados = df.query(f'CH04 == {genero} & ESTADO == 1').shape[0]
    desocupados = df.query(f'CH04 == {genero} & ESTADO == 2').shape[0]
    tasa_desocupacion = desocupados / (ocupados + desocupados)
    return tasa_desocupacion

# Calculando la tasa de desocupación para hombres (CH04 == 1)
tasa_hombres = calcular_tasa_desocupacion(eph_2, 1)
print("Tasa de desocupación para hombres:", tasa_hombres)

# Calculando la tasa de desocupación para mujeres (CH04 == 2)
tasa_mujeres = calcular_tasa_desocupacion(eph_2, 2)
print("Tasa de desocupación para mujeres:", tasa_mujeres)


Tasa de desocupación para hombres: 0.07479247840081314
Tasa de desocupación para mujeres: 0.09144448013705964
